src/buildercore/lifecycle.py

```diff
@@ -50,30 +51,34 @@ def restart(stackname):
 
     history = []
 
-    for node in node_list:
-        push(history, node_states(node_list))
-
-        node.stop()
-        push(history, node_states(node_list))
-
-        node.wait_until_stopped()
-        push(history, node_states(node_list))
-
-        node.start()
-        push(history, node_states(node_list))
-
-        node.wait_until_running()
-        push(history, node_states(node_list))
-
-        node_id = _node_id(node)
-        call_while(
-            lambda: _some_node_is_not_ready(stackname, node=node_id, concurrency='serial'),
-            interval=2,
-            update_msg="waiting for nodes to be networked",
-            done_msg="all nodes have public ips"
-        )
-    update_dns(stackname)
-    return history
+    try:
+        for node in node_list:
+            push(history, node_states(node_list))
+
+            node.stop()
+            push(history, node_states(node_list))
+
+            node.wait_until_stopped()
+            push(history, node_states(node_list))
+
+            node.start()
+            push(history, node_states(node_list))
+
+            node.wait_until_running()
+            push(history, node_states(node_list))
+
+            node_id = _node_id(node)
+            call_while(
+                lambda: _some_node_is_not_ready(stackname, node=node_id, concurrency='serial'),
+                interval=2,
+                update_msg="waiting for nodes to be networked",
+                done_msg="all nodes have public ips"
+            )
+        update_dns(stackname)
+        return history
+    except Exception:
+        LOG.info("Partial restart history of %s: %s", stackname, pformat(history))
```

Comment:
:+1: 

there is a known bug attempting to restart an instance with a multi-az RDS instance - they can't be restarted apparently :(

Reply:
Yes, it's not supported by the API. In this case I stumbled upon a timeout while running a test, and wanted more information next time it happens.